src/version_set.cc

```diff
@@ -204,52 +207,61 @@ Status VersionSet::LogAndApply(VersionEdit* edit) {
   }
   if (!s.ok()) return s;
 
-  return Apply(edit);
+  EditCollector collector;
+  collector.AddEdit(edit);
+  return Apply(&collector);
 }
 
-Status VersionSet::Apply(VersionEdit* edit) {
-  auto cf_id = edit->column_family_id_;
-  auto it = column_families_.find(cf_id);
-  if (it == column_families_.end()) {
-    // TODO: support OpenForReadOnly which doesn't open DB with all column
-    // family so there are maybe some invalid column family, but we can't just
-    // skip it otherwise blob files of the non-open column families will be
-    // regarded as obsolete and deleted.
-    return Status::OK();
-  }
-  auto& files = it->second->files_;
-
-  for (auto& file : edit->deleted_files_) {
-    auto number = file.first;
-    auto blob_it = files.find(number);
-    if (blob_it == files.end()) {
-      fprintf(stderr, "blob file %" PRIu64 " doesn't exist before\n", number);
-      abort();
-    } else if (blob_it->second->is_obsolete()) {
-      fprintf(stderr, "blob file %" PRIu64 " has been deleted before\n",
-              number);
-      abort();
+Status VersionSet::Apply(EditCollector* collector) {
```

Comment:
Change this to `EditCollector&`? RocksDB convention is to use reference for input param, and pointer for output param.